core/databaseDriver.ml

```diff
@@ -79,7 +79,10 @@ module Locator = struct
          loop false pattern (opam_share driver_name)
       | [] -> raise Not_found
       | path :: paths ->
-         match Glob.files path pattern with
+         let files =
```

Comment:
Since 4.08 we can use exception patterns in a nested match, so
```ocaml
match Glob.files path pattern with
| [] -> loop fallback pattern paths
| fs -> fs
| exception Disk.AccessError _ -> []
```